### PR TITLE
Reorganize core.task package

### PR DESCRIPTION
**Pull request by kolloch**
task.tracker did not contain tracking code so I split it into "jobs"
and "update".

### CODE REVIEW

**Comment by kolloch** [comment]
Rebased this on the other PR since I need to continue working on this...

**Comment by kolloch** [comment]
another rebase

File: src/main/scala/mesosphere/marathon/MarathonSchedulerService.scala [hunk 1]
```diff
@@ -69,7 +69,7 @@ class MarathonSchedulerService @Inject() (
   frameworkIdUtil: FrameworkIdUtil,
   @Named(ModuleNames.LEADER_ATOMIC_BOOLEAN) leader: AtomicBoolean,
   appRepository: AppRepository,
-  taskTracker: TaskTracker,
+  taskTracker: TaskTrackerImpl,
```
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/MarathonSchedulerService.scala]: Can we always use a trait and never use impl?

**Comment by aquamatthias** [comment]
I guess I should review only the second commit?

**Comment by gkleiman** [comment]
There's a typo in the commit message: __Reorgines__ core.task package slightly

**Comment by aquamatthias** [comment]
LGTM! Will wait for the original PR to land.

**Comment by kolloch** [comment]
Commit message corrected.

**Comment by kolloch** [comment]
Rebased

**Comment by aquamatthias** [comment]
+1

**Pull request merged**